crates/proof-of-sql/src/sql/parse/error.rs

```diff
@@ -50,50 +50,25 @@ pub enum ConversionError {
     /// General error for invalid expressions
     InvalidExpression(String),
 
-    #[error("Unsupported operation: cannot round decimal: {0}")]
-    /// Decimal rounding is not supported
-    DecimalRoundingError(String),
-
-    #[error("Error while parsing precision from query: {0}")]
-    /// Error in parsing precision in a query
-    PrecisionParseError(String),
-
-    #[error("Decimal precision is not valid: {0}")]
-    /// Decimal precision is an integer but exceeds the allowed limit. We use i16 here to include all kinds of invalid precision values.
-    InvalidPrecision(i16),
-
     #[error("Encountered parsing error: {0}")]
     /// General parsing error
     ParseError(String),
 
+    #[error(transparent)]
+    /// Errors related to decimal operations
+    DecimalConversionError(#[from] DecimalError),
+
+    #[error(transparent)]
+    /// Errors related to decimal operations
+    IntermediateDecimalConversionError(#[from] IntermediateDecimalError),
```

Comment:
Since these are different we need to separate them..or maybe `IntermediateDecimalError` should be included in `DecimalError`?